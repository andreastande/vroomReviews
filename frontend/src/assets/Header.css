.header-container {
  position: relative;
  display: flex;
  justify-content: center;
}

.header-logo {
  position: absolute;
  left: 0px;
  height: 5rem;
  filter: drop-shadow(0 0 10px rgba(255, 0, 0, 0.5));
  z-index: 200;
}

.gray-text-logo {
  filter: none;
}
.gray-text-logo:hover {
  filter: drop-shadow(0 0 5px rgba(255, 255, 255, 0.2));
}

.normal {
  text-decoration: none;
  border-bottom: 0.2rem solid rgb(255, 0, 0);
}

h1 {
  margin: 0;
}

.grey-text {
  opacity: 0.5;
  border-bottom: 0.2rem solid rgba(255, 0, 0, 0);
}

.grey-text:hover {
  opacity: 1;
  border-bottom: 0.2rem solid #ffffff;
}

.pages-menu {
  display: flex;
  justify-content: flex-end;
  border-radius: 1rem;
  gap: 2rem;
  padding: 1rem 0;
  font-size: 1.2rem;
}

.checkbtn {
  margin-top: 1rem;
  display: none;
}

.check {
  display: none;
}

.burger-menu {
  height: 2.5rem;
  margin-right: 0.5rem;
}

@media (orientation: portrait) {
  .header-logo {
    height: 4rem;
  }

  .header-container {
    justify-content: right;
  }

  .checkbtn {
    display: block;
    z-index: 200;
  }

  .pages-menu {
    display: flex;
    flex-direction: column;
    align-items: center;
    position: fixed;
    left: 0;
    width: 100vw;
    height: 85vh;
    border-bottom: 0.2rem solid red;
    justify-content: start;
    background-color: #202020;
    padding-top: 10vh;
    z-index: 100;
    transition: all 0.5s ease-out;
    top: -110vh;
  }

  .check:checked ~ .pages-menu {
    top: 0;
  }
}
