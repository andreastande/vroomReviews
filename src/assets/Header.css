.header-container {
  display: flex;
  justify-content: center;
}

.link-flex {
  display: flex;
}

.normal {
  color: silver;
  text-decoration: none;
  border-bottom: 0.5rem solid #ff0000;
}

.grey-text {
  opacity: 0.6;
  border-bottom: 0.3rem solid silver;
}

.grey-text:hover {
  opacity: 1;
}

.pages-menu {
  display: flex;
  justify-content: flex-end;
  border-radius: 1rem;
  gap: 2rem;
}

.checkbtn {
  margin-top: 1rem;
  display: none;
}

.check {
  display: none;
}

.burger-menu {
  height: 3rem;
}

@media (orientation: portrait) {
  .header-container {
    justify-content: left;
  }

  .checkbtn {
    display: block;
  }

  .pages-menu {
    display: flex;
    flex-direction: column;
    align-items: center;
    position: fixed;
    width: 100%;
    justify-content: center;
    background-color: #202020;
    top: 5rem;
    left: -100%;
    padding: 0;
    z-index: 100;
    text-align: center;
    transition: all 0.5s;
  }

  .check:checked ~ .pages-menu {
    left: 0;
  }
}
