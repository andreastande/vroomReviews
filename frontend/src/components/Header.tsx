import { useState } from 'react';
import { NavLink } from 'react-router-dom';
import '../assets/Header.css';
import logo from '../assets/images/logo.png';
import CloseHamburgerIcon from '../assets/images/Close_Hamburger_icon.png'

const Header = ({
  page,
  setPage,
}: {
  page: string;
  setPage: (page: string) => void;
}) => {
  const [isChecked, setIsChecked] = useState(false);
  const imageUrl = isChecked
  ? CloseHamburgerIcon
  : 'https://www.pngkit.com/png/full/239-2394744_icon-open-nav-icon-white.png';

  // Set page to the page that was selected, and store it in sessionStorage to be able to access it when reloading the page
  function handlePage(page: string) {
    setPage(page);
    sessionStorage.setItem('currentPage', page);
    setIsChecked(false);
  }

  // Toggle the burger menu for mobile
  function toggleCheckbox() {
    setIsChecked(!isChecked);
  }

  return (
    <header className={`header-container ${isChecked ? 'open' : ''}`}>
      <input
        className="check"
        checked={isChecked}
        onChange={toggleCheckbox}
        type="checkbox"
        id="check"
      />
      <label className={`checkbtn ${isChecked ? 'open' : ''}`} htmlFor="check">
        <img
          className="burger-menu"
          src={imageUrl}
          alt="burger-menu"
          style={{
            height: isChecked ? '3rem' : '2.5rem',
            width: isChecked ? '3rem' : '',
          }}
        />
      </label>
      <NavLink to="/project2" onClick={() => handlePage('home')}>
        <img
          className={
            page === 'home'
              ? `header-logo ${isChecked ? 'open' : ''}`
              : `header-logo gray-text-logo ${isChecked ? 'open' : ''}`
          }
          src={logo}
          alt="site-logo"
        />
      </NavLink>
      <nav className="pages-menu">
        <NavLink
          to="/project2/filtercars"
          className={page === 'filter' ? 'normal' : 'normal grey-text'}
          onClick={() => handlePage('filter')}
        >
          <h1>Search</h1>
        </NavLink>
        <NavLink
          to="/project2/favorites"
          className={page === 'favorites' ? 'normal' : 'normal grey-text'}
          onClick={() => handlePage('favorites')}
        >
          <h1>My Favorites</h1>
        </NavLink>
        <NavLink
          to="/project2/reviewedcars"
          className={page === 'reviews' ? 'bottom-link normal' : 'bottom-link normal grey-text'}
          onClick={() => handlePage('reviews')}
        >
          <h1>My Reviews</h1>
        </NavLink>
      </nav>
    </header>
  );
};

export default Header;
