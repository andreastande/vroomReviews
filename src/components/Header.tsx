import { useState } from 'react';
import { Link } from 'react-router-dom';
import '../assets/Header.css';

const Header = () => {
  const [page, setPage] = useState('home');
  const [isChecked, setIsChecked] = useState(false);

  function handlePage(page: string) {
    setPage(page);
    setIsChecked(false);
  }

  function toggleCheckbox() {
    setIsChecked(!isChecked);
  }

  return (
    <div className="header-container">
      <input
        className="check"
        checked={isChecked}
        onChange={toggleCheckbox}
        type="checkbox"
        id="check"
      />
      <label className="checkbtn" htmlFor="check">
        <img
          className="burger-menu"
          src="https://www.pngkit.com/png/full/239-2394744_icon-open-nav-icon-white.png"
          alt=""
        />
      </label>
      <ul className="pages-menu">
        <div className="link-flex">
          <Link
            to="/project2"
            className={page === 'home' ? 'normal' : 'normal grey-text'}
            onClick={() => handlePage('home')}
          >
            <h1>Home</h1>
          </Link>
        </div>
        <div className="link-flex">
          <Link
            to="/project2/filtercars"
            className={page === 'browse' ? 'normal' : 'normal grey-text'}
            onClick={() => handlePage('browse')}
          >
            <h1>Browse cars</h1>
          </Link>
        </div>
        <div className="link-flex">
          <Link
            to="/project2/favorites"
            className={page === 'favorites' ? 'normal' : 'normal grey-text'}
            onClick={() => handlePage('favorites')}
          >
            <h1>Favorites</h1>
          </Link>
        </div>
        <div className="link-flex">
          <Link
            to="/project2/reviewedcars"
            className={page === 'reviews' ? 'normal' : 'normal grey-text'}
            onClick={() => handlePage('reviews')}
          >
            <h1>Your reviews</h1>
          </Link>
        </div>
      </ul>
    </div>
  );
};

export default Header;
