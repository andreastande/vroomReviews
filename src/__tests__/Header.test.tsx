import { render } from "@testing-library/react";
import Header from "../components/Header";
import { test, expect } from "vitest";
import userEvent from "@testing-library/user-event";
import { BrowserRouter } from "react-router-dom";

test("Go to different pages", async () => {
  const { getByText, asFragment } = render(
    <BrowserRouter>
      <Header />
    </BrowserRouter>
  );

  const browse = getByText("Browse cars");
  expect(browse).toBeTruthy(); // The text is found
  await userEvent.click(browse);
  expect(asFragment()).toMatchSnapshot();

  const favorites = getByText("Favorites");
  expect(favorites).toBeTruthy(); // The text is found
  await userEvent.click(favorites);
  expect(asFragment()).toMatchSnapshot();

  const reviews = getByText("Your reviews");
  expect(reviews).toBeTruthy(); // The text is found
  await userEvent.click(reviews);
  expect(asFragment()).toMatchSnapshot();
});