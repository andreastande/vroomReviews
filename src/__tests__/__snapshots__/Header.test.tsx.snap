// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Go to different pages 1`] = `
<DocumentFragment>
  <div
    class="header-container"
  >
    <input
      class="check"
      id="check"
      type="checkbox"
    />
    <label
      class="checkbtn"
      for="check"
    >
      <img
        alt=""
        class="burger-menu"
        src="https://www.pngkit.com/png/full/239-2394744_icon-open-nav-icon-white.png"
      />
    </label>
    <ul
      class="pages-menu"
    >
      <div
        class="link-flex"
      >
        <a
          aria-current="page"
          class="normal grey-text active"
          href="/project2"
        >
          <h1>
            Home
          </h1>
        </a>
      </div>
      <div
        class="link-flex"
      >
        <a
          aria-current="page"
          class="normal active"
          href="/project2/filtercars"
        >
          <h1>
            Browse cars
          </h1>
        </a>
      </div>
      <div
        class="link-flex"
      >
        <a
          class="normal grey-text"
          href="/project2/favorites"
        >
          <h1>
            Favorites
          </h1>
        </a>
      </div>
      <div
        class="link-flex"
      >
        <a
          class="normal grey-text"
          href="/project2/reviewedcars"
        >
          <h1>
            Your reviews
          </h1>
        </a>
      </div>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`Go to different pages 2`] = `
<DocumentFragment>
  <div
    class="header-container"
  >
    <input
      class="check"
      id="check"
      type="checkbox"
    />
    <label
      class="checkbtn"
      for="check"
    >
      <img
        alt=""
        class="burger-menu"
        src="https://www.pngkit.com/png/full/239-2394744_icon-open-nav-icon-white.png"
      />
    </label>
    <ul
      class="pages-menu"
    >
      <div
        class="link-flex"
      >
        <a
          aria-current="page"
          class="normal grey-text active"
          href="/project2"
        >
          <h1>
            Home
          </h1>
        </a>
      </div>
      <div
        class="link-flex"
      >
        <a
          class="normal grey-text"
          href="/project2/filtercars"
        >
          <h1>
            Browse cars
          </h1>
        </a>
      </div>
      <div
        class="link-flex"
      >
        <a
          aria-current="page"
          class="normal active"
          href="/project2/favorites"
        >
          <h1>
            Favorites
          </h1>
        </a>
      </div>
      <div
        class="link-flex"
      >
        <a
          class="normal grey-text"
          href="/project2/reviewedcars"
        >
          <h1>
            Your reviews
          </h1>
        </a>
      </div>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`Go to different pages 3`] = `
<DocumentFragment>
  <div
    class="header-container"
  >
    <input
      class="check"
      id="check"
      type="checkbox"
    />
    <label
      class="checkbtn"
      for="check"
    >
      <img
        alt=""
        class="burger-menu"
        src="https://www.pngkit.com/png/full/239-2394744_icon-open-nav-icon-white.png"
      />
    </label>
    <ul
      class="pages-menu"
    >
      <div
        class="link-flex"
      >
        <a
          aria-current="page"
          class="normal grey-text active"
          href="/project2"
        >
          <h1>
            Home
          </h1>
        </a>
      </div>
      <div
        class="link-flex"
      >
        <a
          class="normal grey-text"
          href="/project2/filtercars"
        >
          <h1>
            Browse cars
          </h1>
        </a>
      </div>
      <div
        class="link-flex"
      >
        <a
          class="normal grey-text"
          href="/project2/favorites"
        >
          <h1>
            Favorites
          </h1>
        </a>
      </div>
      <div
        class="link-flex"
      >
        <a
          aria-current="page"
          class="normal active"
          href="/project2/reviewedcars"
        >
          <h1>
            Your reviews
          </h1>
        </a>
      </div>
    </ul>
  </div>
</DocumentFragment>
`;
